### PR TITLE
Add policy to store the latest messages

### PR DESCRIPTION
**Pull request by jcmonteiro**
Add policies to store the most recent received messages (based on their timestamp or on the order of arrival); These policies **do not** call any callback automatically. It is up to the programmer to call the `call` method when it is time to process the information.

These policies are meant to be used in networks where information is processed periodically.

### CODE REVIEW

**Comment by ZhenshengLee** [comment]
> It is up to the programmer to call the `call` method when it is time to process the information.

@jcmonteiro 

Could you provide some docs to explain the motivation of this policy? or more info about why, when and how we should use this policy?

File: include/message_filters/sync_policies/latest_stamped.h [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+*  BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+*  LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+*  CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+*  LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+*  ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+*  POSSIBILITY OF SUCH DAMAGE.
+*********************************************************************/
+
+#ifndef MESSAGE_FILTERS__SYNC_LatestStamped_H_
+#define MESSAGE_FILTERS__SYNC_LatestStamped_H_
```
Comment (clalancette) [inline include/message_filters/sync_policies/latest_stamped.h:36]:
```suggestion
#ifndef MESSAGE_FILTERS__SYNC_LATESTSTAMPED_H_
#define MESSAGE_FILTERS__SYNC_LATESTSTAMPED_H_
```

**Review by clalancette — CHANGES_REQUESTED** [review]
I know this is really old, so sorry for the long delay in looking at it.  If you'd like to give up on getting this in, I'd totally understand.

If you'd still like to move forward, I've left some stuff inline to fix.  Also, I'm not a huge fan of the terminology "latest" to describe this policy.  While it is accurate, I don't think it will be hugely meaningful to people, and seems too close to "latest_time" (which we already have).  Maybe we can rename it to "user controlled", or something like that.

**Comment by ZhenshengLee** [comment]
@jcmonteiro  Friendly ping for updates.

**Comment by ZhenshengLee** [comment]
@jcmonteiro Would you agree that I continue to work with your branch and complete with the PR? 

I'll begin with following the change requests in this PR, then create a new PR with the forked branch with my commits.

**Comment by jcmonteiro** [comment]
> @jcmonteiro Would you agree that I continue to work with your branch and complete with the PR?
> 
> I'll begin with following the change requests in this PR, then create a new PR with the forked branch with my commits.

Hi @ZhenshengLee. I've been out of the loop of the robotics and ROS communities for a while now. Please feel free to take this PR in the direction that you see fit.

**Comment by ZhenshengLee** [comment]
Close this and continue with #107

**Pull request closed without merging**